pkg/storage/vault/aws/aws.go

```diff
@@ -0,0 +1,135 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	awsConfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+	"github.com/scratchdata/scratchdata/config"
+)
+
+type AWSVault struct {
+	client *secretsmanager.Client
+	prefix string
+}
+
+func NewAWSVault(conf map[string]any, destinations []config.Destination) (*AWSVault, error) {
+	cfg, err := awsConfig.LoadDefaultConfig(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	var accessKey, secretKey, prefix, region string
+
+	if accessKeyVal, ok := conf["AccessKey"].(string); ok {
+		accessKey = accessKeyVal
+	} else {
+		return nil, errors.New("AccessKey not found or not a string")
+	}
+
+	if secretKeyVal, ok := conf["SecretKey"].(string); ok {
+		secretKey = secretKeyVal
+	} else {
+		return nil, errors.New("SecretKey not found or not a string")
+	}
+
+	if prefixVal, ok := conf["Prefix"].(string); ok {
+		prefix = prefixVal
+	} else {
+		return nil, errors.New("prefix not found or not a string")
+	}
+
+	if regionVal, ok := conf["Region"].(string); ok {
+		region = regionVal
+	} else {
+		return nil, errors.New("region not found or not a string")
+	}
+
+	if accessKey != "" && secretKey != "" {
+		cfg.Credentials = credentials.NewStaticCredentialsProvider(accessKey, secretKey, "")
+	}
+
+	cfg.Region = region
+	client := secretsmanager.NewFromConfig(cfg)
+
+	vault := &AWSVault{
+		client: client,
+		prefix: prefix,
+	}
+
+	for _, dest := range destinations {
```

Comment:
There is no need to upload destinations to AWS. Either credentials will be read from a config file, or they will be read from AWS, depending on how the vault is configured.